File: Ooproject/src/Application.java
import java.util.Scanner;

public class Application {
    public void run() {
        int userInput = 1;
        var a = new Number(1);
        var b = new Number(2);
        var c = new Number(1);
        while (userInput != 0) {
            showUserOptions();
            Scanner scanner = new Scanner(System.in);
            userInput = scanner.nextInt();
            if (userInput == 0) break;
            switch (userInput) {
                case 1:
                    System.out.println("Input coefficient a: ");
                    a.number = scanner.nextDouble();
                    System.out.println("Input coefficient b: ");
                    b.number = scanner.nextDouble();
                    System.out.println("Input coefficient c: ");
                    c.number = scanner.nextDouble();
                    break;
                case 2:
                    var rootCalcPolynomial = new Polynomial(a, b, c);
                    Number[] roots = rootCalcPolynomial.calcRoots();
                    System.out.println("roots[0] = " + roots[0].number + ", roots[1] = " + roots[1].number);
                    break;
                case 3:
                    var outputPolynomial = new Polynomial(a, b, c);
                    outputPolynomial.showPolynomial();
                    break;
                default:
                    break;
            }
        }
    }

    private static void showUserOptions() {
        System.out.println("1) Input polynomial coefficients");
        System.out.println("2) Roots");
        System.out.println("0) Exit");
    }
}

File: application.cpp
#include "application.h"
#include <iostream>
#include "number.h"
#include "polinom.h"

using namespace std;

application::application()
{

}

int application::exec()
{
    int ch = 1, n;
    number a = 1, b = 2, c = 1;
    number x[2];
    while (ch!=0)
    {
        menu();
        cin>>ch;
            if (ch == 0) break;
        switch (ch)
        {
        case 1:
            cout<<"a= ";
            cin>>a;
            cout<<"b= ";
            cin>>b;
            cout<<"c= ";
            cin>>c;
            break;
        case 2:
        {
            polinom p(a, b, c);
            n = p.roots(x);
            cout<<x[0]<<' '<<x[1]<<endl;
        }
            break;
        case 3:
        {
            polinom p(a, b, c);
            cout<<"p(x) = "<<p;
        }
            break;
        default:
            break;
        }
    }
    cout<<"exec"<<endl;
    return 0;
}

void application::menu()
{
    cout<<"1. polinom coeff"<<endl;
    cout<<"2. roots"<<endl;
    cout<<"0. exit"<<endl;
}
